### ISSUE TITLE
How to connect controller with sink.sky in Cooja emulator

### ISSUE THREAD

**Issue by hieu1204** [sdnwiselab/sdn-wise-contiki#3]
Dear Mr.Milardo,
I can run controller connecting to SDN-WISE java emulated node, but I don't know how to connect controller with sink.sky node ( sdn-wise-contiki/sdn-wise/cooja_firmwares/ ) ( if it's possible ) .
It would be great if you can guide me how it work.
Thank you.

**Comment by escuderoDP** [comment]
Hi, did you solve the problem?
I used the "Serial Socket (CLIENT ...)" tool using port 9991, but the controller displays an error message.

**Comment by hieu1204** [comment]
@escuderoDP As far as I know, the sdn-wise-java controller is used ONLY for java emulated node. The serial connection of the sink.sky node is basically different. I had done it by creating a new controller interacting with the sink.sky node via the serial interface (in cooja). The basic idea of such connection you can find here
http://anrg.usc.edu/contiki/index.php/Collect_View
Take a look at _Relevant Directories_

**Comment by escuderoDP** [comment]
Thanks for the answer, I saw that the controller works only with the Java mote, but I did not understand how you applied Collect View as a crontolator. Can you describe the scenario and procedure you did? Please! I'm also using COOJA. I've used Collect View in other applications by inserting its source code.

**Comment by escuderoDP** [comment]
Thank you very much.

**Comment by escuderoDP** [comment]
Good evening. I found this implementation of yours, but I have not tested it yet. Here you have a controller already programmed and created for sink.sky according to the previous instructions?

https://github.com/hieu1204/sdn-wise-trickle/tree/master/sdnwise-controller

**Comment by hieu1204** [comment]
@escuderoDP yes, this is a simple controller which provides a visual network topology and sends routing paths to the nodes

**Comment by escuderoDP** [comment]
@hieu1204 Should something appear in "GraphStream"?

https://imgur.com/a/JFaFbO7

**Comment by hieu1204** [comment]
@escuderoDP Just try again and let me know, a topology should appear.

**Comment by escuderoDP** [comment]
@hieu1204  Good night. Sorry for the inconvenience, but I want to make your application work for me. I had the following problems:
* The controller is not sending any message;
* The sink prints "inconsistent detected";
* Consequently GraphStream does not appear anything.
I replaced the folder cooja and added the controller folder in contiki/tools, as instructed by you.

https://imgur.com/a/OFWFxV8

**Comment by hieu1204** [comment]
@escuderoDP could you send me your csc file as well as firmwares that you used. I will try to find the problem and update the tutorial.

**Comment by basmahassan123** [comment]
This link doesn't work. Please,re-appload it again. I need this programmed controller


https://github.com/hieu1204/sdn-wise-trickle/tree/master/sdnwise-controller

**Comment by hieu1204** [comment]
@basmahassan123 here it is https://gitlab.com/hieu.nq/sdn-wise-trickle.git
The automatic build process using ant is not completed yet, so if you need to rebuild the controller, just using a Java IDE instead.

**Comment by RaHeron95** [comment]
> Good evening. I found this implementation of yours, but I have not tested it yet. Here you have a controller already programmed and created for sink.sky according to the previous instructions?
> 
> https://github.com/hieu1204/sdn-wise-trickle/tree/master/sdnwise-controller

Hi Iḿ interested also in your implementation, @escuderoDP  said that was some instructions or tutorial to implement, could you help me to indicate where are those in the new link repository 

> @basmahassan123 here it is https://gitlab.com/hieu.nq/sdn-wise-trickle.git
> The automatic build process using ant is not completed yet, so if you need to rebuild the controller, just use a Java IDE instead.
> Update: github repo https://github.com/hieu1204/sdn-wise-trickle
